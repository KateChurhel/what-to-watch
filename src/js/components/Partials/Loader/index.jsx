// libraries
import React from 'react';

const Loader = ({ isActive, children }) => {
  const containerClass = isActive ? 'loader-overlay-container' : '';

  return (
    <div className={containerClass}>
      {children}
      {isActive && (
        <div className="loader-spinner-wrapper">
          <div className="loader-spinner" />
        </div>
      ) }
    </div>
  );
};

export default Loader;
