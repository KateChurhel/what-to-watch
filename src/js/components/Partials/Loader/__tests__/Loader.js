import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Loader from '../index';

describe('Loader: ', () => {
  it('correctly renders with loader', () => {
    const tree = shallow(<Loader isActive>Test content</Loader>);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('correctly renders without loader', () => {
    const tree = shallow(<Loader>Test content</Loader>);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
