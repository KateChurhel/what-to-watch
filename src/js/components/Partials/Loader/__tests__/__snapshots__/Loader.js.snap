// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader:  correctly renders with loader 1`] = `
<div
  className="loader-overlay-container"
>
  Test content
  <div
    className="loader-spinner-wrapper"
  >
    <div
      className="loader-spinner"
    />
  </div>
</div>
`;

exports[`Loader:  correctly renders without loader 1`] = `
<div
  className=""
>
  Test content
</div>
`;
